src/cargo.rs

```diff
@@ -29,7 +30,12 @@ fn cargo(project: &Project) -> Command {
 }
 
 pub fn build_dependencies(project: &Project) -> Result<()> {
-    let _ = cargo(project).arg("generate-lockfile").status();
+    let workspace_cargo_lock = path!(project.workspace / "Cargo.lock");
+    if workspace_cargo_lock.exists() {
+        let _ = fs::copy(workspace_cargo_lock, path!(project.dir / "Cargo.lock"));
+    } else {
+        let _ = cargo(project).arg("generate-lockfile").status();
```

Comment:
Cargo.lock should always exists when trybuild is executed. When the real project was build, cargo will always generate a Cargo.lock, if there doesn't exists any.